src/clover/mains_supply/__utils__.py

```diff
@@ -104,6 +105,18 @@ def get_intermittent_supply_status(  # pylint: disable=too-many-locals
                 else:
                     status.append(0)
         times = pd.DataFrame(status)
+
+        # Identify indices where there is no power
+        zero_indices = times.index[times.iloc[:, 0] == 0]
+
+        # Extract the corresponding outage time
+        outage_time = grid_attributes.iloc[zero_indices % 24, 1]
+        outage_time.index = zero_indices
+
+        # Append additional zeros to the grid availability data frame
+        for i, n in zip(zero_indices, outage_time):
```

Comment:
## :bug: Is this  a bug :question:
1. I'm not sure whether `.iloc` works in this way? Worth checking with `pdb` that the dataframe looks like what you expect;
2. Are these indicies right? I'd check by using `pdb` as I think starting from `i+1` will fill in hours from then on, but ending at `i+n+1` will likely fill in an extra hour, no?

## :memo: `i, n` -> nicer names?
E.G., `outage_time` and `length_of_outage`? It makes the next line much easier to read, for one thing:
```python
times.iloc[outage_time + 1 : outage_time + length_of_outage + 1, 0]
```